apps/api-gateway/src/services/custom_report_service.py

```
@@ -0,0 +1,556 @@
+        columns: List[Dict],
+        user_id: str,
+        description: Optional[str] = None,
+        filters: Optional[Dict] = None,
+        sort_by: Optional[List[Dict]] = None,
+        default_format: str = ReportFormat.XLSX,
+        is_public: bool = False,
+        store_id: Optional[str] = None,
+    ) -> ReportTemplate:
+        """创建报表模板"""
+        if data_source not in DATA_SOURCE_FIELDS:
+            raise ValueError(f"不支持的数据源: {data_source}，可选: {list(DATA_SOURCE_FIELDS.keys())}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)